src/bridge/command.rs

```diff
@@ -57,14 +60,27 @@ fn create_platform_shell_command(command: &str, args: &[&str]) -> StdCommand {
 
         result
     } else if cfg!(target_os = "macos") {
-        let shell = env::var("SHELL").unwrap_or_else(|_| "/bin/sh".to_string());
-        let mut result = StdCommand::new(shell);
+        let user =
+            env::var("USER").unwrap_or_explained_panic("USER environment variable not found");
+        let shell = env::var("SHELL").unwrap_or_else(|_| "/bin/zsh".to_string());
+        let shell_name = shell.rsplit('/').next().unwrap();
 
-        if env::var_os("TERM").is_none() {
-            result.arg("-l");
-        }
-        result.arg("-c");
-        result.arg(format!("{} {}", command, args.join(" ")));
+        // Exec the shell with argv[0] prepended by '-' so it becomes a login shell.
+        // `login` normally does this itself, but `-l` disables this.
+        let exec = format!("exec -a -{} {} {}", shell_name, command, args.join(" "));
+
+        // See "man login". It sets up some important env vars like $PATH and $HOME.
+        // On macOS, use the `login` command so it will appear as a tty session.
+        let mut result = StdCommand::new("/usr/bin/login");
+
+        // We use a special flag to tell login not to prompt us for a password, because we're
+        // going to spawn it as the current user anyway. The addition of "p",
+        // preserves the environment.
+        // -f: Bypasses authentication for the already-logged-in user.
+        // -l: Skips changing directory to $HOME and prepending '-' to argv[0].
+        // -p: Preserves the environment.
+        // -q: Forces quiet logins, as if a .hushlogin is present.
+        result.args(["-flpq", &user, &shell, "-lc", &exec]);
```

Comment:
I think we should use `/bin/zsh` here like in Alacritty. The user shell does not necessarily have the exec command, so it would fail for those shells. There should also be no need to run that as a login shell (we don't need to do it twice), so `-c` is enough.

Reply:
Sure, I am actually against about forcing our users into a specific shell. It means they will be obliged to maintain the .zshenv which is something we shouldn't.

I'll remove the `exec` and launch as their own shell.

Also exec is not really needed here to make it work.

Reply:
> There should also be no need to run that as a login shell (we don't need to do it twice), so -c is enough.

thank you for remembering this but unfortunately it didn't load the correct PATH for both shells I've tested using only `-c`, it is basically not loading their shell configuration accordingly with `-l`

I also agree it's redundant but seems like alacrity is doing the same to make it work.

Reply:
done